src/b-splines/constant.jl

```diff
@@ -5,8 +5,25 @@ struct Nearest <: ConstantInterpType end
 struct Previous <: ConstantInterpType end
 struct Next <: ConstantInterpType end
 
-struct Constant{T<:ConstantInterpType} <: Degree{0} end
-Constant() = Constant{Nearest}()
+struct Constant{T<:ConstantInterpType,BC<:Union{Throw{OnGrid},Periodic{OnCell}}} <: DegreeBC{0}
+    bc::BC
```

Comment:
Is `bc` stateful here? Or could we always use `BC()` to construct it at runtime?

Reply:
https://github.com/JuliaMath/Interpolations.jl/blob/a8370bd2b4ef77cfe689c55c3ed217ab528f69c1/src/b-splines/b-splines.jl#L142
I am not sure I understand the question, but I think since the above line uses `deg.bc`, so `bc` should be a field.